examples/sensor-websocket/src/components/SensorWebSocket.jsx

```diff
@@ -0,0 +1,206 @@
+          {sensor.name}
+        </span>
+        <Col className="p-0">
+          <ListGroup className="py-1 my-2">
+            {sensor_item.map(listItem)}
+          </ListGroup>
+        </Col>
+      </Col>
+    </Row>
+  );
+};
+
+const ConnectionStatus = props => {
+  const connection_type = props.connection_type;
+  return (
+    <Row className="m-3">
+      <Col className="col-12 text-dark px-3 py-2 bg-transparent border-0 w-100 text-left btn text-decoration-none">
+        <h6 className="m-0 font-weight-bold position-relative">
+          Device Status : &nbsp;
+          <span
+            className={`status-tag rounded-circle d-inline-block mr-2 ${
+              connection_type ? "connected" : "disconnected"
+            }`}
+          />
+          {connection_type ? "Connected" : "Disconnected"}
+        </h6>
+      </Col>
+    </Row>
+  );
+};
+
+class SensorWebSocket extends Component {
+  state = {
+    sensors: {},
+    device: {
+      connection_type: false
+    },
+    socket: null,
+    connected: false,
+    alerts: []
+  };
+
+  setDeviceStatus = () => {
+    const { deviceID } = this.props;
+    getDeviceDataById(deviceID).then(response => {
+      this.setState({ device: { connection_type: response.connected } });
+    });
+  };
+
+  socketDataHandler = response => {
+    if (response.event === "new_event") {
+      const { sensors } = this.state;
+      const time = response.payload.timestamp;
+      const event = response.payload.event;
+      if (event.path) {
+        sensors[event.path] = {
+          value: event.value,
+          name: event.path.split("/")[1],
+          //"/${device_id}/value".split("/") => [ "", "${device_id}", "value" ]
+          timestamp: moment(time).format(getLocaleFormat())
+        };
+      }
+      this.setState({ sensors });
+    }
+  };
+
+  onSocketError = () => {
+    const { alerts } = this.state;
+    alerts.unshift({ type: "info", msg: "an error occurred" });
+    this.setState({ alert });
+  };
+
+  onSocketClose = () => {
+    const { socket, alerts } = this.state;
+    socket.disconnect();
+    alerts.unshift({ type: "info", msg: "this socket is closed." });
+    this.setState({ alert });
+  };
+
+  onSocketOpen = () => {
+    const { alerts } = this.state;
+    alerts.unshift({ type: "success", msg: "the socket is open." });
+    this.setState({ alert });
+  };
+
+  connectSocket = () => {
+    const { deviceID } = this.props;
+    const socketUrl = getWSUrl();
+    const phoenixSocket = new Socket(socketUrl, socketParams);
+    phoenixSocket.onError(this.onSocketError);
+    phoenixSocket.onClose(this.onSocketClose);
+    phoenixSocket.onOpen(this.onSocketOpen);
+    phoenixSocket.onMessage(this.socketDataHandler);
+    phoenixSocket.connect();
+    this.channelConnect(phoenixSocket, deviceID);
+  };
+
+  channelConnect = (phoenixSocket, deviceID) => {
+    const channel_token = getChannelToken();
+    const channel = phoenixSocket.channel("rooms:example:test", {
```

Comment:
if all your rooms have the same name, when you open the same app in 2 browser tabs, all the triggers will interfere with each other, and of course this is not intended for this particular App.